ext/standard/math.c

```diff
@@ -875,11 +877,39 @@ PHPAPI int _php_math_basetozval(zval *arg, int base, zval *ret)
 			c -= 'A' - 10;
 		else if (c >= 'a' && c <= 'z')
 			c -= 'a' - 10;
-		else
+		else if (isspace(c)) {
+			if (num == 0 && fnum == 0) {
+				continue;
+			}
+			lastchar = 1;
 			continue;
-
-		if (c >= base)
+		} else {
+			invalidchars++;
 			continue;
+		}
+
+		if (lastchar == 1 && c > 0 && c < base) {
+			invalidchars++;
+		}
+		lastchar = 0;
+
+
+		if (c >= base) {
+			if (*(s-2) == '0' && num == 0) {
```

Comment:
This `*(s-2)` may perform a logically out of bounds read if we're at the start of the string. It will still be part of the zend_string, but a byte in the hash value, which may cause misbehavior.

Again I would recommend to move this outside of this loop and to a prefix check at the start.